app/models/session.rb

```diff
@@ -33,7 +42,14 @@ def secure_random_token(length = DEFAULT_TOKEN_LENGTH)
   end
 
   def setup_defaults
-    @token ||= secure_random_token unless persisted?
```

Comment:
`def setup_defaults` gets called after both `Session.new` and `Session.find`.  The `unless persisted?` part looks like a way for the method to have logic that applies only to the former.  It seems redundant though because of our use of the ||= operator and the fact that `@token` would never be nil due to `validates :token, presence: true`